affected:
- package:
    ecosystem: PyPI
    name: tensorflow-cpu
    purl: pkg:pypi:tensorflow-cpu
  ranges:
  - events:
    - introduced: '0'
    - fixed: 480641e3599775a8895254ffbc0fc45621334f68
    repo: https://github.com/tensorflow/tensorflow
    type: GIT
  - events:
    - introduced: '0'
    - fixed: 2.2.0rc0
    - introduced: 2.2.0
    - fixed: 2.3.0rc0
    - introduced: 2.3.0
    - fixed: 2.3.4
    - introduced: 2.4.0
    - fixed: 2.4.3
    type: ECOSYSTEM
aliases:
- CVE-2021-29551
- GHSA-vqw6-72r7-fgw7
details: TensorFlow is an end-to-end open source platform for machine learning. The
  implementation of `MatrixTriangularSolve`(https://github.com/tensorflow/tensorflow/blob/8cae746d8449c7dda5298327353d68613f16e798/tensorflow/core/kernels/linalg/matrix_triangular_solve_op_impl.h#L160-L240)
  fails to terminate kernel execution if one validation condition fails. The fix will
  be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow
  2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also
  affected and still in supported range.
id: PYSEC-0000-CVE-2021-29551
modified: '2021-08-27T03:22:30.499582Z'
published: '2021-05-14T20:15:00Z'
references:
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/480641e3599775a8895254ffbc0fc45621334f68
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-vqw6-72r7-fgw7
